### PR TITLE
RDM-4345 - add retry for ES index verification

### PR DESCRIPTION
**Pull request by gbenadikar**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-4345

### Change description ###
Add retries to ES index verification in case ES has been setup in asynchronous mode

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/helper/elastic/ElasticsearchTestDataLoaderExtension.java [hunk 1]
```diff
@@ -162,4 +167,19 @@ private void waitUntilLogstashIndexesCaseData(long millis) {
...
+
+    private boolean verifyIndex(String indexAlias) {
+        try {
+            asElasticsearchApiUser()
+                .when()
+                .get(indexAlias)
+                .then()
+                .statusCode(200);
+        } catch (AssertionError e) {
+            LOG.warn("Retrying Elasticsearch index api due to error: {}", e.getMessage());
```
Comment (mario-paniccia) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/tests/helper/elastic/ElasticsearchTestDataLoaderExtension.java]: shall we log as INFO?

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/util/RetryUtility.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.ccd.datastore.tests.util;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import static org.awaitility.Awaitility.with;
+
+public class RetryUtility {
```
Comment (mario-paniccia) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/tests/util/RetryUtility.java]: shall we call this TestUtils so we can add more utility methods in the future?

Comment (gbenadikar) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/tests/util/RetryUtility.java]: done

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/helper/elastic/ElasticsearchTestDataLoaderExtension.java (hunk 1 above)
Comment (gbenadikar) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/tests/helper/elastic/ElasticsearchTestDataLoaderExtension.java]: done

**Pull request merged**